test/ocfclient/main.go

```diff
@@ -0,0 +1,546 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	enjson "encoding/json"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/plgd-dev/kit/codec/json"
+	"github.com/plgd-dev/sdk/app"
+	"github.com/plgd-dev/sdk/local"
+	"github.com/plgd-dev/sdk/test"
+)
+
+const Timeout = time.Second * 8
+
+type (
+	// Ocfclient for working with devices
+	Ocfclient struct {
+		localClient        *local.Client
+		secureDevices	  []local.DeviceDetails
+	}
+)
+
+type testSetupSecureClient struct {
+	ca      []*x509.Certificate
+	mfgCA   []*x509.Certificate
+	mfgCert tls.Certificate
+}
+
+func (c *testSetupSecureClient) GetManufacturerCertificate() (tls.Certificate, error) {
+	if c.mfgCert.PrivateKey == nil {
+		return c.mfgCert, fmt.Errorf("not set")
+	}
+	return c.mfgCert, nil
+}
+
+func (c *testSetupSecureClient) GetManufacturerCertificateAuthorities() ([]*x509.Certificate, error) {
+	if len(c.mfgCA) == 0 {
+		return nil, fmt.Errorf("not set")
+	}
+	return c.mfgCA, nil
+}
+
+func (c *testSetupSecureClient) GetRootCertificateAuthorities() ([]*x509.Certificate, error) {
+	if len(c.ca) == 0 {
+		return nil, fmt.Errorf("not set")
+	}
+	return c.ca, nil
+}
+
+func NewTestClient() *local.Client {
+	appCallback, err := app.NewApp(nil)
+	if err != nil {
+		panic(err)
+	}
+	c, err := local.NewClientFromConfig(&local.Config{
+		KeepAliveConnectionTimeoutSeconds: 1,
+		ObserverPollingIntervalSeconds:    1,
+	}, appCallback, test.NewIdentityCertificateSigner, func(error) {})
+	if err != nil {
+		panic(err)
+	}
+	return c
+}
+
+func NewTestSecureClient() (*local.Client, error) {
```

Comment:
Just New Client , It's not a test, and security is a default thing. It evokes it can be also unsecure, and that's bad :)